### ISSUE TITLE
Enhancement: [prefer-optional-chain] Also rewrite `if (x) x.something()`

### ISSUE THREAD

**Issue by fregante** [typescript-eslint/typescript-eslint#6309]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-optional-chain/

### Description

The rule apparently only tries to prevent `a && a.b`, but another way to write that is `if (a) a.b`

Could the rule be updated to include `if` conditions?

### Fail

```typescript
if (callback) {
  callback();
}

if (list) {
  list.push(item);
}
```

### Pass

```typescript
callback?.();

list?.push(item);

// Optional chaining doesn't apply here
if (callback) {
  callbacks.push(callback);
}
```

**Comment by bradzacher** [comment]
This would be a great addition - it's a really common pattern that could be simplified. Though I can see this being an option - I can see some people prefer the more verbose form.

Though it's worth noting that this would be non-trivial to build and handle all the cases.

**Comment by Josh-Cena** [comment]
What about multiple statements?

```ts
if (foo) {
  foo.bar();
  foo.baz();
}
```

It could be simplified but some would see it as "non-performant".

**Comment by fregante** [comment]
I think that it'd make sense as an auto fix of a single statement, but not multiple.

**Comment by JoshuaKGoldberg** [comment]
It's not technically safe to auto-fix multiple statements. An earlier statement might modify the value being optionally chained.

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgGIHt0HF1gQCzmQG8AoZZAIzigAoBKALmQDd1gATAbnKrgC8GzNpx4BfUqQA2EMMhiZmGbLgJEAPsgCuIDhBigIHZAF4SvanXrmKFBelPbd%2Bw915iANBYEMSyCRKkAPRByABC%2BuhQEIykwDDItPbWZHaYAHSWDDxp6Jk%2B9OKSIcgAgjCQUKT2APz5Vjy1%2BYKFQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

* Before: _`Cannot read properties of undefined (reading 'baz')`_
* After: no crash

One shudders to imagine the use case for wanting to preserve the thrown error, but it does technically change code behavior.

**Comment by kirkwaiblinger** [comment]
> It's not technically safe....

[See also: ill-behaved getters](https://www.typescriptlang.org/play/?target=99&ts=5.7.0-dev.20240904#code/JYOwLgpgTgZghgYwgAgJrANYQM7IN4BQyyMA9qQFz5HHIBGcUAFAJRUBupwAJgNw0BfGsgA+yAK4huEGKAjcCQggBsIYZAE9MOKuiy4AvMiatkBgHzViq9WVIBxNZCgBhOMuWHkABn7W1yKAg0GZWtAxQVKYWYbSBMMZ2jmDObh64ADzIAEwssXHIYAAWUKQA7sjBFQCiUKXMAEQAIjwgAOTqAOYBdsgQIKTinUWFwAC2OACEDSx+BQLIwrQIpCDYpKoAdMqknUwN7sqBuGUQHtOzS0LE18hQauJQIPndtuSmhAUk5Mmph9gAagBczi9zAj2eQWgIOQtyEAhYrAIBGACSYWn0mzseU+mm02Cx5E2EVY-CUAHpyXjMXYAPzExikoA) 😁 

I agree with not rewriting multistatement optional chains, _but_, I think that in general we should follow TS's lead in control flow modelling, where a function call, or, god forbid, a member access, does not invalidate narrowing.
